src/Chapter3.hs

```diff
@@ -374,6 +383,42 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Knight = MkKnight
+    { knightHealth  :: Int
+    , knightAttack  :: Int
+    , knightGold    :: Int
+    }
+
+data Monster = MkMonster
+    { monsterHealth  :: Int
+    , monsterAttack  :: Int
+    , monsterGold    :: Int
+    }
+
+
+fight :: Knight -> Monster -> Int
+fight knight monster
+    | (knightAttack knight) >= (monsterHealth monster) = (knightGold knight) + (monsterGold monster)
+    -- is there any way to access records' fields in a more succint way ? like a dot notation or something ?
+    | otherwise =
+        case (knightHealth knight) < (monsterAttack monster) of
+          True -> (-1)
+          False -> (knightGold knight)
```

Comment:
You can write it as two other guards:
```suggestion
    | knightHealth knight < monsterAttack monster = -1
    | otherwise = knightGold knight
```